R/html_dependencies.R

```diff
@@ -194,16 +194,26 @@ html_reference_path <- function(path, lib_dir, output_dir) {
 # return the html dependencies as an HTML string suitable for inclusion
 # in the head of a document
 html_dependencies_as_string <- function(dependencies, lib_dir, output_dir) {
+  # Only copy deps with path
+  dependencies_href <- filter_dependencies(dependencies, "href")
+  dependencies <- filter_dependencies(dependencies, "file")
 
   if (!is.null(lib_dir)) {
     dependencies <- lapply(dependencies, copyDependencyToDir, lib_dir)
     dependencies <- lapply(dependencies, makeDependencyRelative, output_dir)
   }
-  return(renderDependencies(dependencies, "file", encodeFunc = identity,
+  #return(
+  tags <- renderDependencies(dependencies, "file", encodeFunc = identity,
                             hrefFilter = function(path) {
                               html_reference_path(path, lib_dir, output_dir)
                             })
-  )
+  #)
+  HTML(paste0(c(tags, renderDependencies(dependencies_href, "href")), collapse = "\n"))
```

Comment:
I believe the current implementation means that all `href` deps will be loading **after** `file` deps. 
I would think **rmarkdown** should not touch the order of dependencies. Is it safe to change the order ? 
Could there not be side effect ?

Reply:
Looking at this now, I can't see where the makeDependencyRelative / copyDependency logic is reimplemented. 
I'll have to check this out properly

Reply:
& I agree about the order, I will probably modify filter_dependencies to take a vector of `src_types` to cover both types at once.

Reply:
Doing this https://github.com/rstudio/rmarkdown/pull/1948/files#r608469286 should keep the order of dependencies. Trick is to use `mustWork = FALSE` in htmltools functions